### PR TITLE
Refactored the loss curves and loss maps

### PR DESCRIPTION
**Pull request by micheles**
This is needed for https://github.com/gem/oq-risklib/issues/490. The demos are running here: https://ci.openquake.org/job/zdevel_oq-risklib/1162

### CODE REVIEW

**Comment by ptormene** [comment]
Demos did not pass

**Comment by micheles** [comment]
I have fixed the demos.

File: openquake/calculators/event_based_risk.py [hunk 1]
```diff
@@ -312,12 +307,18 @@ def post_execute(self, result):
...
+                    if o == RC:
+                        rcurves[lt][:, r] = poes
+                    elif insured_losses:
+                        icurves[lt][:, r] = poes
                     saved[self.outs[o]] += poes.nbytes
                 self.datastore.hdf5.flush()
 
         self.datastore['avg_losses-rlzs'] = avg_losses
         saved['avg_losses-rlzs'] = avg_losses.nbytes
+        self.datastore['rcurves-rlzs'] = rcurves
```
Comment (ptormene) [inline openquake/calculators/event_based_risk.py]: Please check if this requires `if rcurves` or if it's fine as it is

**Comment by ptormene** [comment]
Apart from a minor doubt, LGTM

File: openquake/calculators/event_based_risk.py (hunk 1 above)
Comment (micheles) [inline openquake/calculators/event_based_risk.py]: No, this is fine, rcurves are always generated, the optional curves are the icurves.

**Pull request merged**